### PR TITLE
Fix some build warnings, and update mongo references 

### PR DESCRIPTION
**Pull request by sakash279**
This change addresses:
1. Update Mongo references to use newer dlls, that are compatible with VSTS to produce binaries.
2. Fix some warnings that get generated by AutoGeneratedBindingRedirects in certain projects.

### CODE REVIEW

File: AzureTable/Microsoft.DataTransfer.AzureTable.FunctionalTests/Microsoft.DataTransfer.AzureTable.FunctionalTests.csproj [hunk 1]
```diff
@@ -66,6 +66,18 @@
       <HintPath>..\..\packages\Microsoft.Data.Services.Client.5.8.3\lib\net40\Microsoft.Data.Services.Client.dll</HintPath>
       <Private>True</Private>
     </Reference>
+    <Reference Include="Microsoft.OData.Core, Version=7.5.1.20914, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\packages\Microsoft.OData.Core.7.5.1\lib\portable-net45+win8+wpa81\Microsoft.OData.Core.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Microsoft.OData.Edm, Version=7.5.1.20914, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (nahk-ivanov) [inline AzureTable/Microsoft.DataTransfer.AzureTable.FunctionalTests/Microsoft.DataTransfer.AzureTable.FunctionalTests.csproj:76]: Is this supposed to replace Microsoft.Data.Edm on line 57 or they both are needed? Same for Microsoft.Data.OData - if I understand correctly, this is a new version of that library/package. Do we still need both? There might be some conflicts between dependencies (if they have any shared dependencies with different versions, like JSON.NET, for example).

File: MongoDb/Microsoft.DataTransfer.MongoDb.FunctionalTests/packages.config [hunk 2]
```diff
@@ -1,8 +1,11 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="Castle.Core" version="3.3.3" targetFramework="net452" />
-  <package id="MongoDB.Bson" version="2.2.4" targetFramework="net452" />
-  <package id="MongoDB.Driver" version="2.2.4" targetFramework="net452" />
-  <package id="MongoDB.Driver.Core" version="2.2.4" targetFramework="net452" />
+  <package id="DnsClient" version="1.0.7" targetFramework="net452" />
```
Comment (nahk-ivanov) [inline MongoDb/Microsoft.DataTransfer.MongoDb.FunctionalTests/packages.config:4]: You need to review all new libraries you are adding/will be redistributing. Depending on the license, they may require you to update third-party notices as well as perform an IP scan on everything we plan to redistribute in pre-compiled form.

**Review by nahk-ivanov — CHANGES_REQUESTED** [review]
Make sure all new third-party binaries are cleared with legal.

File: MongoDb/Microsoft.DataTransfer.MongoDb.FunctionalTests/packages.config (hunk 2 above)
Comment (sakash279) [reply MongoDb/Microsoft.DataTransfer.MongoDb.FunctionalTests/packages.config:4]: Ack. I've submitted a request to perform an IP scan on the open source packages being pulled here. The only additional library added here for documentation purposes is DnsClient, which is a reference for MongoDb.

File: AzureTable/Microsoft.DataTransfer.AzureTable.FunctionalTests/Microsoft.DataTransfer.AzureTable.FunctionalTests.csproj (hunk 1 above)
Comment (sakash279) [reply AzureTable/Microsoft.DataTransfer.AzureTable.FunctionalTests/Microsoft.DataTransfer.AzureTable.FunctionalTests.csproj:76]: you're correct, they're both present in the build output previously, so I am adding them to not break any use-cases. 
They're both dependencies to Microsoft.Azure.CosmosDB.Table.2.0.0 (newer version)

File: MongoDb/Microsoft.DataTransfer.MongoDb.FunctionalTests/packages.config (hunk 2 above)
Comment (sakash279) [reply MongoDb/Microsoft.DataTransfer.MongoDb.FunctionalTests/packages.config:4]: The review has been completed, and all scan results have come back. I will take necessary actions to update licenses.

**Comment by sakash279** [comment]
> Make sure all new third-party binaries are cleared with legal.

Done

**Pull request merged**